### PR TITLE
RE2022-116: Set selection marks in genome attribs on selection cretion

### PR DESCRIPTION
**Pull request by MrCreosote**

### CODE REVIEW

File: src/service/selection_processing.py [hunk 1]
```diff
@@ -2,17 +2,60 @@
...
+    deps: PickleableDependencies,
+    args: list[Any]
+):
+    hb = None
+    arangoclient = None
+    try:
+        arangoclient, storage = await deps.get_storage()
+        async def heartbeat(millis: int):
+            await storage.send_selection_heartbeat(internal_selection_id, millis)
+        hb = processing.Heartbeat(heartbeat, processing.HEARTBEAT_INTERVAL_SEC)
```
[the remark this answers is not in the document]
Comment (dauglyon) [reply src/service/selection_processing.py:32]: this is likely a me-problem but a comment might be helpful

[the remark this answers is not in the document]
Comment (dauglyon) [reply src/service/selection_processing.py:32]: is it so the asyncio loop only runs when selection stuff is actively running?

[the remark this answers is not in the document]
Comment (MrCreosote) [reply src/service/selection_processing.py:32]: So if the process just dies, the next time the process is accessed the service will see the process isn't complete, the heartbeat is old, and restart it

[the remark this answers is not in the document]
Comment (MrCreosote) [reply src/service/selection_processing.py:32]: Otherwise the selection is stuck in limbo forever

[the remark this answers is not in the document]
Comment (dauglyon) [reply src/service/selection_processing.py:32]: ahhhhh that makes more sense, thx

**Review by dauglyon — APPROVED** [review]

**Pull request merged**